File: Famous Algos/Binary Search/BinarySearch.c
#include <stdio.h>

int binarySearch(int* array, int sizeOfArray, int targetNum) {
    int startPointer = 0, endPointer = sizeOfArray - 1, midPointer, i = 0;
    int found = 0;

    for (i = 0; i < sizeOfArray; i++) {
        midPointer = (startPointer + endPointer) / 2;

        if (array[midPointer] == targetNum) {
            found = 1;
            break;
        } else {
            if (array[midPointer] > targetNum) {
                endPointer = midPointer - 1;
            } else {
                startPointer = midPointer + 1;
            }
        }
    }
    return found;
}

int main() {
    int arr[] = {1, 2, 3, 4, 5};
    int targetNum = 0;
    int sizeOfArray = sizeof(arr) / sizeof(int);
    int result = binarySearch(arr, sizeOfArray, targetNum);
    printf("%d", result);
    return 0;
}

File: Famous Algos/Binary Search/BinarySearch.rs
fn binary_search(array: &[i32], target_number: i32) -> bool {
    let mut start_pointer = 0;
    let mut end_pointer = array.len() - 1;
    let mut found = false;

    while start_pointer <= end_pointer {
        let mid_pointer = (start_pointer + end_pointer) / 2;
        if array[mid_pointer] == target_number {
            found = true;
            break;
        }
        else if array[mid_pointer] > target_number {
            end_pointer = mid_pointer - 1;
        } else {
            start_pointer = mid_pointer + 1;
        }
    }

    return found;
}

fn main() {
    let result = binary_search(&[1, 2, 3, 4, 5, 6, 7], 1);
    println!("{}", result)
}
